client/state/data-layer/wpcom/sites/comments/replies/new/index.js

```diff
@@ -4,20 +4,29 @@
 import { COMMENTS_WRITE } from 'state/action-types';
 import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
 import {
+	hasParentCommentId,
 	dispatchNewCommentRequest,
 	updatePlaceholderComment,
 	handleWriteCommentFailure,
 } from 'state/data-layer/wpcom/sites/utils';
 
 export const writePostComment = ( { dispatch }, action ) => {
-	if ( ! action.parentCommentId ) {
+	if ( ! hasParentCommentId( action ) ) {
 		return;
 	}
 
 	const path = `/sites/${ action.siteId }/comments/${ action.parentCommentId }/replies/new`;
 	dispatchNewCommentRequest( dispatch, action, path );
 };
 
+const onSuccess = ( { dispatch }, action, ...args ) => ! hasParentCommentId( action )
+	? updatePlaceholderComment( { dispatch }, action, ...args )
+	: null;
+
+const onFailure = ( { dispatch }, action, ...args ) => ! hasParentCommentId( action )
+	? handleWriteCommentFailure( { dispatch }, action, ...args )
+	: null;
+
 export default {
-	[ COMMENTS_WRITE ]: [ dispatchRequest( writePostComment, updatePlaceholderComment, handleWriteCommentFailure ) ]
+	[ COMMENTS_WRITE ]: [ dispatchRequest( writePostComment, onSuccess, onFailure ) ]
```

Comment:
I abhors duplication as nature abhors vacuum, and tried very hard not to add another action for replies to comments. But I agree that it would be a lot clearer. If @dmsnell agrees, I'll move forward with adding a new action, the corresponding action creator, and updating the components.